src/test/java/com/ordestiny/tdd/kata/DiamondTest.java

```diff
@@ -0,0 +1,47 @@
+package com.ordestiny.tdd.kata;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.Arrays;
+import org.junit.jupiter.api.Test;
+
+public class DiamondTest {
+
+
+
+	@Test
+	public void buildDiamond_whenWidestPointCharIsNull_returnException() {
+		assertThrows(IllegalArgumentException.class, () -> Diamond.buildDiamond('1'));
+	}
+
+	@Test
+	public void buildDiamond_widestPointCharSetToA_returnA() {
+		assertEquals(Diamond.buildDiamond('A'), "A");
+	}
+
+	@Test
+	public void buildDiamond_widestPointCharSetToCharB_returnCorrectDiamondString() {
+		assertEquals(Diamond.buildDiamond('B'), "ABBA");
+	}
+
+	@Test
+	public void buildDiamond_widestPointCharSetToCharC_returnCorrectDiamondString() {
+		assertEquals(Diamond.buildDiamond('C'), "ABBCCCBBA");
```

Comment:
the logic of buildDiamond_widestPointCharSetToA_returnA, buildDiamond_widestPointCharSetToCharB_returnCorrectDiamondString and buildDiamond_widestPointCharSetToCharC_returnCorrectDiamondString methods are exactly the same.

I think you can use parameterized test instead??

Reply:
Thanks for your suggestion @januslinhc . I've updated the PR.